src/main/java/net/minecraftforge/common/capabilities/CapabilityManager.java

```diff
@@ -61,31 +63,38 @@
         Objects.requireNonNull(storage,"Attempted to register a capability with no storage implementation");
         Objects.requireNonNull(factory,"Attempted to register a capability with no default implementation factory");
         String realName = type.getName().intern();
-        if (providers.containsKey(realName)) {
-            LOGGER.error(CAPABILITIES, "Cannot register capability implementation multiple times : {}", realName);
-            throw new IllegalArgumentException("Cannot register a capability implementation multiple times : "+ realName);
-        }
+        Capability<T> cap;
+
+        synchronized (providers)
+        {
+            if (providers.containsKey(realName)) {
+                LOGGER.error(CAPABILITIES, "Cannot register capability implementation multiple times : {}", realName);
+                throw new IllegalArgumentException("Cannot register a capability implementation multiple times : "+ realName);
+            }
 
-        Capability<T> cap = new Capability<>(realName, storage, factory);
-        providers.put(realName, cap);
+            cap = new Capability<>(realName, storage, factory);
```

Comment:
Extremely minor, but we can be optimistic and create the Capability object outside the `synchronized` block to minimize the time spent inside this block.